src/modules/launcher/Wox.Infrastructure/Image/ImageCache.cs

```diff
@@ -22,7 +23,26 @@ public class ImageCache
                 var i = _data[path];
                 return i;
             }
-            set { _data[path] = value; }
+            set 
+            {
+                _data[path] = value;
+
+                // To prevent the dictionary from drastically increasing in size by caching images, the dictionary size is not allowed to grow more than the permissibleFactor * maxCached size
+                // This is done so that we don't constantly perform this resizing operation and also maintain the image cache size at the same time
+                if (_data.Count > permissibleFactor * MaxCached)
```

Comment:
[clarification] How frequently do we hit `MaxCached` this limit? We are doing a clean up each time this happens, which seems to be a costly operation.

Reply:
As mentioned in my previous comment to Arjun, the maximum time it takes to do this is 10 ms (because the size of the dictionary is small). On trying it out on my machine, I did not see any performance issue. Maybe you can try it out and let me know if I should increase it to maybe 200. Right now it’s hit on every 100 unique image icons and it’s not very often. Cannot quantify a value because it depends on what the user searches for.

Reply:
I didn't notice any visible difference but I think we should be able to support caching more images. Current code limits image size to 64*64 pixel bitmap image (~ 20KB) , so even with caching 1000 images we will never cross 20MB threshold.

Reply:
It's about preference. Do we want to cache 1000 images and use 20 MB of memory when we can rather cache just 50 images without any significant performance hit? I just tried to profile the reordering of the dictionary and it took around 24 ms with maxCached as 1000 on my system in comparison to 10 ms for 50.Feel free to correct me here but I think it's not about it being under 20 MB, rather about why we want to use 20 MB when we can get away with a few hundred KB. Let me know what you think. 

@ryanbodrug-microsoft, @crutkas do you have any preferences over the number of images to cache? Right now it is capped at 100, so whenever the images cached increase over 100, it is resized to 50 (trying to emulate the behavior of a vector).